depthai_ros_driver/include/depthai_ros_driver/ros_agnostic.hpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef DEPTHAI_ROS_DRIVER__ROS_AGNOSTIC_HPP
+#define DEPTHAI_ROS_DRIVER__ROS_AGNOSTIC_HPP
+
+#if defined(USE_ROS2)
+  #include <rclcpp/node.hpp>
+  #define ROS_LOGGER(...) RCUTILS_LOG_WARN_NAMED("depthai_ros", __VA_ARGS__)
+using ROSNodeHandle = std::shared_ptr<rclcpp::Node>;
+using RosTime = rclcpp::Time;
+using RosDuration = rclcpp::Duration;
+#else
+  #include <ros/ros.h>
+  #define ROS_LOGGER(...) ROS_WARN_NAMED("depthai_ros", __VA_ARGS__)
+using ROSNodeHandle = std::shared_ptr<ros::NodeHandle>;
+using RosTime = ros::Time;
+using RosDuration = ros::Duration;
+#endif
+
+namespace rr {
+
+/*
+ * Ros agnostic is a group of ROS wrapper classes, which enables agnostic ROS
+ * implementation of both ros1 and ros2. This can be switched during compile
+ * by defining 'USE_ROS2' in the Cmake file
+ */
+namespace ros_agnostic {
+
+//==============================================================================
+class Publisher;
+class Subscription;
+class Service;
+class Timer;
```

Comment:
Thoughts on adding the declarations for these classes in the header file as well?

Reply:
The detailed declaration is made in `/depthai_ros_driver/impl/ros_agnostic.hpp` to make it less cluttered. I can also move it here if this is favorable.

Reply:
No the definitions, just declaration. cc: @gbiggs

Reply:
Addressed https://github.com/osrf/depthai_ros/pull/4/commits/c04922d9f1b691c8fd58c29c0b0c7abc286c45a4